waku/v2/rendezvous/rendezvous.go

```diff
@@ -99,8 +111,40 @@ func (r *Rendezvous) Start(ctx context.Context) error {
 const registerBackoff = 200 * time.Millisecond
 const registerMaxRetries = 7
 
-func (r *Rendezvous) getRandomServer() *rendezvousPoint {
-	return r.rendezvousPoints[rand.Intn(len(r.rendezvousPoints))] // nolint: gosec
+func (r *Rendezvous) getRandomRendezvousPoint(ctx context.Context) <-chan *rendezvousPoint {
+	var dialableRP []*rendezvousPoint
+	now := time.Now()
+	for _, rp := range r.rendezvousPoints {
+		if now.After(rp.NextTry()) {
+			dialableRP = append(dialableRP, rp)
+		}
+	}
+
+	result := make(chan *rendezvousPoint, 1)
+
+	if len(dialableRP) > 0 {
+		result <- r.rendezvousPoints[rand.Intn(len(r.rendezvousPoints))] // nolint: gosec
+	} else {
+		if len(r.rendezvousPoints) > 0 {
```

Comment:
If there is no dialable RP, choosing one randomly is fine if discover is done by Waku layer. 
Due to #581 , discovery triggers are moved to application layer..in that case, it would be possible for an application to spam the RP even if by mistake by constantly invoking discover. By having this backoff strategy, in a way we are preventing same RP not being dialed constantly.
Maybe, instead of choosing to dial if no dialable RPs are there, it would be better to return a wait-retry to the application based on which the app can redial again.

Another thought that comes to me, is do we even need to define this backoff strategy or just let application handle it?

Reply:
> Maybe, instead of choosing to dial if no dialable RPs are there, it would be better to return a wait-retry to the application based on which the app can redial again.

Hmm, not sure I follow. The code is set so when there is no dialable RP, it will then wait until the RP with the shortest wait time is available. Dialing will be only attempted when there is a RP 

> Another thought that comes to me, is do we even need to define this backoff strategy or just let application handle it?

Good point. I'm thinking that i could refactor this code to extracted part of the logic into a function that an application could use to avoid having to implement their own backoff strategy from scratch. What do you think?

Reply:
> > Maybe, instead of choosing to dial if no dialable RPs are there, it would be better to return a wait-retry to the application based on which the app can redial again.
> 
> Hmm, not sure I follow. The code is set so when there is no dialable RP, it will then wait until the RP with the shortest wait time is available. Dialing will be only attempted when there is a RP
> 
My bad, got confused when i was looking at diff only in github. When i looked in vsCode, i understood.
> > Another thought that comes to me, is do we even need to define this backoff strategy or just let application handle it?
> 
> Good point. I'm thinking that i could refactor this code to extracted part of the logic into a function that an application could use to avoid having to implement their own backoff strategy from scratch. What do you think?
Yeah, that's a good idea since anyways you have coded the backoff strategy logic.